odxtools/database.py

```diff
@@ -44,7 +44,7 @@ def __init__(self,
 
             tmp = [
                 read_diag_layer_container_from_odx(dlc_el, enable_candela_workarounds=enable_candela_workarounds) \
-                  for dlc_el in dlc_elements
+                  for dlc_el in filter(None, dlc_elements)
```

Comment:
to avoid confusion, please take care of this using an `if` on line 43, i.e.,

```python
dlc = root.find("DIAG-LAYER-CONTAINER")
if dlc is not None:
  dlc_elements.append(dlc)
```

Reply:
(note for a future version: add support for non-DLC files...)